crates/noirc_evaluator/src/ssa/acir_gen/internal_var.rs

```diff
@@ -38,9 +38,15 @@ impl InternalVar {
     pub(crate) fn set_id(&mut self, id: NodeId) {
         self.id = Some(id)
     }
+    pub(crate) fn get_id(&self) -> Option<NodeId> {
+        self.id
+    }
     pub(crate) fn cached_witness(&self) -> &Option<Witness> {
         &self.cached_witness
     }
+    pub(crate) fn set_witness(&mut self, w: Option<Witness>) {
+        self.cached_witness = w;
```

Comment:
Can we put an assert here about the previous value of `self.cached_witness` here?

Reply:
I put a debug_assert since setting a witness twice is not a case for failing.